changes:

vikyw89 opened the pull request:
when visiting / 

- signed in user will be redirected to contacts/ 
- unsigned in user will be redirected to login/

when visiting login 
- signed in user will be redirected to contacts/

vikyw89 committed:
changes:
- when visiting /
    - signed in user will be redirected to contacts/
    - unsigned in user will be redirected to login/
- when visiting login
    - signed in user will be redirected to contacts/

vikyw89 committed:
fix: lint

weaponsforge commented on client/src/pages/index.js:
```diff
@@ -11,7 +13,16 @@ const defaultState = {
+    } else {
+      router.push('/login')
+    }
+  })
```
Looks good, but I found a few inconsistencies for the case:

> when visiting /
> - signed in user will be redirected to contacts/

Please view the attached video and take note of the multiple redirects on the URL (URL route goes from "/" to "/login" and finally, "/contacts"), and the red-letter warnings on the chrome dev console.

### Video: Multiple Redirection and Chrome Dev Warnings

https://user-images.githubusercontent.com/56998001/235442888-9adff0da-13e5-4e9b-903b-d8dd6df362ed.mp4

---

### Suggestions

Here are some suggestions but please confirm and verify if it's effective:

Listen for the `auth.authLoading` status in the deps array and call `router.push` only after `auth.authLoading` becomes `false`.

```
  useEffect(()=>{
    if (!auth.authLoading) {
      if (auth.authUser) {
        router.push('/contacts')
      } else {
        router.push('/login')
      }
    }
  }, [auth])
```

weaponsforge commented on client/src/pages/index.js:
```diff
@@ -11,7 +13,16 @@ const defaultState = {
+    } else {
+      router.push('/login')
+    }
+  })
```
Looks good, but I found a few inconsistencies for the case:

> when visiting /
> - signed in user will be redirected to contacts/

Please view the attached videos (taken while running this branch) and take note of the multiple redirects (URL route goes from "/" to "/login" and finally, "/contacts"), and the red-letter warnings on the chrome dev console.

### Video: Multiple Redirection and Chrome Dev Warnings

When visiting "/" while signed in

https://user-images.githubusercontent.com/56998001/235442888-9adff0da-13e5-4e9b-903b-d8dd6df362ed.mp4

### Video: Chrome Dev Warnings

When visiting "/" while signed-out

https://user-images.githubusercontent.com/56998001/235443795-9e687fbb-e0ae-41da-a959-c6a611e9de9f.mp4

---

### Suggestions

Here are some suggestions to minimize the mentioned bugs and warnings but please confirm and verify if it's effective:

Listen for the `auth.authLoading` status in the deps array and call `router.push` only after `auth.authLoading` becomes `false`.

```
  useEffect(()=>{
    if (!auth.authLoading) {
      if (auth.authUser) {
        router.push('/contacts')
      } else {
        router.push('/login')
      }
    }
  }, [auth])
```

vikyw89 replied:
[Screencast from 2023-05-02 12-45-27.webm](https://user-images.githubusercontent.com/112059651/235582153-4f677aa4-79fc-402e-bd60-46e3ff1af04e.webm)
- moved loading to separate components, listening to change to
    - contacts loading, auth loading
    - useRouter() change of url
- temporarily removed loading true in [doc_id] route, to prevent infinite loading
- fix console error on login/ and contacts/ and / route

weaponsforge replied:
Thanks for the updates @vikyw89 ! I re-run the branch and verified that the multiple redirects and mentioned console errors are fixed with your latest updates.

I would like to note one small thing, which I will mention in a separate review comment. This PR looks ready to merge after confirming this item.

vikyw89 committed:
Merge branch 'dev' of https://github.com/weaponsforge/my-phonebook into fix/viky-132

vikyw89 committed:
- updated useSyncV, better snippet and autoComplete for config
- moved loading to separate components, listening to change to
    - contacts loading
    - useRouter() change of url
- temporarily removed loading true in [doc_id] route, to prevent infinite loading
- fix console error on login/ and contacts/ and / route

weaponsforge commented on client/src/common/ui/loadingLinear/index.js:
```diff
@@ -1,12 +1,39 @@
+import { updateSyncV, useSyncV } from 'use-sync-v'
 
 export const LoadingLinear = () => {
+  const authLoading = useSyncV('auth.loading')
```
Currently, we do not have an `auth.loading` use-sync-v store field value. Using it like this always assigns an `undefined` value to the `authLoading` variable.

Kindly see the use-sync-v `auth` store fields definition in the `/client/src/lib/store/index.js` file for reference. If you meant to listen for the Firebase Auth "loading" status, pls use `auth.authLoading`.

#### P.S. You can rename

`auth.authLoading` => `auth.loading` if this sounds more consistent, but please be sure to update all `auth.authLoading` references in the codes.

vikyw89 committed:
fix: auth.authLoading
